mods/sfinv/api.lua

```diff
@@ -0,0 +1,167 @@
+sfinv = {
+	pages = {},
+	pages_unordered = {},
+	homepage_name = "sfinv:crafting",
+	contexts = {},
+	enabled = true
+}
+
+function sfinv.register_page(name, def)
+	if not name or not def or not def.get then
```

Comment:
But he'd still have to write 12 lines of code to distinguish between any of the possible failed input values. And a mod developer should be able to understand and debug failed asserts :)

There's nothing wrong with `error()` as a function, but it shouldn't replace `assert()` where asserts could do the job IMHO.